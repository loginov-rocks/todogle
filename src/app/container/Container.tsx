import {
  AppBar, Drawer, IconButton, Toolbar, Typography,
} from '@material-ui/core';
import { Menu } from '@material-ui/icons';
import * as React from 'react';
import { Route, Switch } from 'react-router-dom';

import * as R from '../../routes';
import { fetchTaskLists } from '../../store/actions';
import { useDispatch } from '../../store/dispatch';

import CreateTaskList from './createTaskList/CreateTaskList';
import DrawerContent from './drawerContent/DrawerContent';
import Home from './home/Home';
import Task from './task/Task';
import TaskList from './taskList/TaskList';

import styles from './Container.module.css';

const Container = () => {
  const dispatch = useDispatch();
  const [isOpenOnMobile, setIsOpenOnMobile] = React.useState(false);

  React.useEffect(() => {
    dispatch(fetchTaskLists());
  }, [dispatch]);

  const handleDrawerToggle = () => {
    setIsOpenOnMobile(!isOpenOnMobile);
  };

  return (
    <>

      <nav>

        <Drawer
          classes={{
            paper: styles.drawerPaper,
          }}
          className={styles.drawerMobile}
          onClose={handleDrawerToggle}
          open={isOpenOnMobile}
          variant="temporary"
        >
          <DrawerContent onMenuItemClick={handleDrawerToggle} />
        </Drawer>

        <Drawer
          classes={{
            paper: styles.drawerPaper,
          }}
          className={styles.drawerDesktop}
          open
          variant="permanent"
        >
          <DrawerContent />
        </Drawer>

      </nav>

      <div className={styles.wrapper}>

        <AppBar position="static">
          <Toolbar>
            <IconButton
              className={styles.drawerToggle}
              color="inherit"
              edge="start"
              onClick={handleDrawerToggle}
            >
              <Menu />
            </IconButton>
            <Typography component="h1" variant="h6">
              Todogle
            </Typography>
          </Toolbar>
        </AppBar>

        <main>
          <Switch>
            <Route path={R.CREATE_TASK_LIST}>
              <CreateTaskList />
            </Route>
            <Route path={R.TASK}>
              <Task />
            </Route>
            <Route path={R.TASK_LIST}>
              <TaskList />
            </Route>
            <Route path={R.HOME}>
              <Home />
            </Route>
          </Switch>
        </main>

      </div>

    </>
  );
};

export default Container;
