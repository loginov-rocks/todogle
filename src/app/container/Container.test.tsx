import { shallow } from 'enzyme';
import * as React from 'react';

import Container from './Container';

it('matches snapshot', () => {
  const wrapper = shallow(
    <Container />,
  );

  expect(wrapper).toMatchSnapshot();
});
