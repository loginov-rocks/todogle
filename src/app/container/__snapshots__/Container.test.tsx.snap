// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Fragment>
  <nav>
    <WithStyles(ForwardRef(Drawer))
      className="drawerMobile"
      classes={
        Object {
          "paper": "drawerPaper",
        }
      }
      onClose={[Function]}
      open={false}
      variant="temporary"
    >
      <DrawerContent
        onMenuItemClick={[Function]}
      />
    </WithStyles(ForwardRef(Drawer))>
    <WithStyles(ForwardRef(Drawer))
      className="drawerDesktop"
      classes={
        Object {
          "paper": "drawerPaper",
        }
      }
      open={true}
      variant="permanent"
    >
      <DrawerContent />
    </WithStyles(ForwardRef(Drawer))>
  </nav>
  <div
    className="wrapper"
  >
    <WithStyles(ForwardRef(AppBar))
      position="static"
    >
      <WithStyles(ForwardRef(Toolbar))>
        <WithStyles(ForwardRef(IconButton))
          className="drawerToggle"
          color="inherit"
          edge="start"
          onClick={[Function]}
        >
          <MenuIcon />
        </WithStyles(ForwardRef(IconButton))>
        <WithStyles(ForwardRef(Typography))
          component="h1"
          variant="h6"
        >
          Todogle
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Toolbar))>
    </WithStyles(ForwardRef(AppBar))>
    <main>
      <Switch>
        <Route
          path="/create-task-list"
        >
          <CreateTaskList />
        </Route>
        <Route
          path="/task-list/:taskListId/task/:id"
        >
          <Task />
        </Route>
        <Route
          path="/task-list/:id"
        >
          <TaskList />
        </Route>
        <Route
          path="/"
        >
          <Home />
        </Route>
      </Switch>
    </main>
  </div>
</Fragment>
`;
